src/VisualStudio/IntegrationTest/TestSetup/TestTraceListener.cs

```diff
@@ -91,7 +91,14 @@ public override void WriteLine(string? message, string? category)
 
         private static void Exit(string? message)
         {
-            FatalError.ReportAndPropagate(new Exception(message));
+            if (message?.Contains("Pretty-listing introduced errors in error-free code") ?? false)
+            {
+                // Ignore this known assertion failure
+                FatalError.ReportAndCatch(new Exception(message), ErrorSeverity.Critical);
+                return;
+            }
+
+            FatalError.ReportAndPropagate(new Exception(message), ErrorSeverity.Critical);
```

Comment:
The only difference between ReportAndCatch and ReportAndPropagate is what it returns for use in an exception filter. So this isn't doing what you think it's doing.

Reply:
(blocking _only_ for this bit, since this will mean exceptions you might expect are being thrown aren't)

Reply:
➡️ This is now corrected (assertion failures will lead to test failures)